dask/dataframe/io/sql.py

```diff
@@ -104,7 +107,15 @@ def read_sql_table(
     if index_col is None:
         raise ValueError("Must specify index column to partition on")
     engine_kwargs = {} if engine_kwargs is None else engine_kwargs
-    engine = sa.create_engine(uri, **engine_kwargs)
+    if isinstance(con, str):
+        uri = con
+        engine = sa.create_engine(uri, **engine_kwargs)
+    elif isinstance(con, Engine):
+        engine = con
+        uri = str(engine.url)
+    else:
+        raise ValueError("Invalid DB `con`: %s" % con)
```

Comment:
It's an engine, not a "`con`", whatever that is. 
Would be nice to see this without use of `isinstance`.

Reply:
renamed to `engine`